lib/elixir/lib/enum.ex

```diff
@@ -2066,6 +2066,12 @@ defmodule Enum do
     end
   end
 
+  # TODO: Remove me on v2.0
+  def min_max(%{__struct__: Range, first: first, last: last} = range, empty_fallback) do
```

Comment:
I believe we don't need this clause because it will fallback to the protocol.

Reply:
```
iex(7)> Enum.min_max(old_range, 9)                                           
** (FunctionClauseError) no function clause matching in Enum.min_max/2    
    
    The following arguments were given to Enum.min_max/2:
    
        # 1
        10..1//-1
    
        # 2
        9
    
    Attempted function clauses (showing 2 out of 2):
    
        def min_max(first..last//step = range, empty_fallback) when is_function(empty_fallback, 0)
        def min_max(enumerable, empty_fallback) when is_function(empty_fallback, 0)
    
    (elixir 1.12.2) lib/enum.ex:2057: Enum.min_max/2
```

Reply:
This one is failing because the second argument you are passing is invalid, not because of the range.